packages/core/src/components/essentials/title-meter.ts

```diff
@@ -0,0 +1,184 @@
+		const title = svg.selectAll("text.meter-title")
+			.data([dataset[groupMapsTo]]);
+
+		title.enter()
+			.append("text")
+			.classed("meter-title", true)
+			.merge(title)
+			.attr("x", 0)
+			.attr("y", "1em")
+			.text(d => d);
+
+		title.exit().remove();
+
+		// appends the associated percentage after title
+		this.appendPercentage();
+
+		// if status ranges are provided (custom or default), display indicator
+		this.displayStatus();
+
+		// get the max width of a title (with consideration for the status/percentage)
+		const maxWidth = this.getMaxTitleWidth();
+		const titleElement = DOMUtils.appendOrSelect(svg, "text.meter-title");
+
+		if (maxWidth > 0 && titleElement.node().getComputedTextLength() > maxWidth) {
+			this.truncateTitle(titleElement, maxWidth);
+		}
+	}
+
+	/**
+	 * Appends the corresponding status based on the value and the peak.
+	 */
+	displayStatus() {
+		const self = this;
+		const svg = this.getContainerSVG();
+		const options = this.model.getOptions();
+
+		const containerBounds = DOMUtils.getSVGElementSize(
+			this.services.domUtils.getMainSVG(),
+			{ useAttr: true }
+		);
+
+		// need to check if the width is 0, and try to use the parent attribute
+		// this can happen if the chart is toggled on/off and the height is 0 for the parent, it wont validateDimensions
+		const containerWidth = containerBounds.width ? containerBounds.width : this.parent.node().getAttribute("width");
+
+		// get the status from the model
+		const status = this.model.getStatus();
+		const radius = Tools.getProperty(options, "meter", "status", "indicatorSize") / 2;
+
+		// create a group for the icon/inner path
+		const statusGroup = DOMUtils.appendOrSelect(svg, `g.status-indicator`)
+			.classed(`status--${status}`, status != null)
+			.attr("transform", `translate(${containerWidth - radius}, 0)`);
+
+		const data = status ? [status] : [];
+		const icon = statusGroup.selectAll("circle.status")
+			.data(data);
+
+		icon
+			.enter()
+			.append("circle")
+			.merge(icon)
+			.attr("class", "status")
+			.attr("r", radius)
+			.attr("cx", 0)
+			.attr("cy", `calc(1em / 2)`);
+
+		const innerIcon = statusGroup.selectAll("path.innerFill")
+			.data(data);
+
+		innerIcon.enter()
+			.append("path")
+			.merge(innerIcon)
+			.attr("d", self.getStatusIconPathString(status))
+			.attr("transform", `translate(-${radius}, 0)`)
+			.attr("class", "innerFill");
+
+		innerIcon.exit().remove();
+		icon.exit().remove();
+	}
+
+	/**
+	 * Appends the associated percentage to the end of the title
+	 */
+	appendPercentage() {
+		const dataValue = this.model.getDisplayData().value;
+
+		// use the title's position to append the percentage to the end
+		const svg = this.getContainerSVG();
+		const title = DOMUtils.appendOrSelect(svg, "text.meter-title");
+
+		// check if it is enabled
+		const data = Tools.getProperty(this.model.getOptions(), "meter", "title", "percentageIndicator", "enabled") === true ?
```

Comment:
Would this work?
```suggestion
		const data = Tools.getProperty(this.model.getOptions(), "meter", "title", "percentageIndicator", "enabled") ?
```
or
```suggestion
		const data = !!Tools.getProperty(this.model.getOptions(), "meter", "title", "percentageIndicator", "enabled") ?
```

Reply:
It's just better typed this way, since many other values will pass with your proposed way
![image](https://user-images.githubusercontent.com/14989804/85038746-f55b6c00-b154-11ea-8641-94e96b5a6f7c.png)

Reply:
Okay